### PR TITLE
Restart smwingsd on failure

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5453

### CODE REVIEW

**Review by DavidePrincipi — COMMENTED** [review]
If the failure is permanent does it try forever or gives up after some attempts?

**Comment by stephdl** [comment]
https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=

 the doc states 

> Setting this to on-failure is the recommended choice for long-running services, in order to increase reliability by attempting automatic recovery from errors.

I have no informations that the restart action stops after a number of attempts

**Comment by DavidePrincipi** [comment]
Yes now I found them from your link:

> Note that service restart is subject to unit start rate limiting configured with StartLimitIntervalSec= and StartLimitBurst=, see systemd.unit(5) for details. A restarted service enters the failed state only after the start limits are reached.

3 failed starts in 20 seconds could be reasonable...

**Comment by stephdl** [comment]
oups your right :')

**Review by DavidePrincipi — APPROVED** [review]

File: root/usr/lib/systemd/system/smwingsd.service [hunk 1]
```diff
@@ -8,5 +8,6 @@ Type=forking
 ExecStart=/usr/libexec/nethserver/smwingsd
 ExecStop=
 KillMode=process
-
-
+Restart=on-failure
+StartLimitBurst=3
```
Comment (DavidePrincipi) [inline root/usr/lib/systemd/system/smwingsd.service:12]: Move to [unit] section?

**Review by DavidePrincipi — CHANGES_REQUESTED** [review]

File: root/usr/lib/systemd/system/smwingsd.service (hunk 1 above)
Comment (stephdl) [reply root/usr/lib/systemd/system/smwingsd.service:12]: yes, seems so, let me check

**Review by DavidePrincipi — APPROVED** [review]
As @stephdl pointed out, the docs do not refer to our systemd version, 219. `StartLimitBurst` and `StartLimitInterval` were moved to `[Unit]`, but are still acceptable under `[Service]` too.

https://lists.freedesktop.org/archives/systemd-devel/2017-July/039255.html

**Pull request merged**